### PR TITLE
feat: app bridge theme

### PR DESCRIPTION
**Pull request by anxobotana**

### CODE REVIEW

File: packages/app-bridge-theme/src/AppBridgeTheme.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    context<Key extends keyof Context>(key: Key): ContextReturn<Context, Key>;
+    context(key?: keyof Context | void): unknown;
+
+    subscribe<EventName extends keyof Event>(
+        eventName: EventNameParameter<EventName, Event>,
+        callback: EventCallbackParameter<EventName, Event>,
+    ): EventUnsubscribeFunction;
+
+    /**
+     * Search in the current Guideline for a given query.
```
Comment (mike85) [inline packages/app-bridge-theme/src/AppBridgeTheme.ts:36]: Please deprecate this function as the Audi Theme uses this. We’re removing it once they can style the Frontify search dialog.

File: packages/app-bridge-theme/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+export type TemplateContext = { templateId: string; type: ThemeTemplate } & (
+    | { type: 'documentPage'; document: GuidelineDocument; documentPage: GuidelineDocumentPage }
+    | { type: 'library'; document: GuidelineDocumentLibrary }
+    | { type: 'cover'; coverPage: GuidelineCoverPage }
+);
+
+export type State = {
+    settings: Record<string, unknown>;
+};
```
Comment (mike85) [inline packages/app-bridge-theme/src/types.ts:23]: We can remove this

File: packages/app-bridge-theme/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    portalToken: string | null;
+    currentLanguage: string;
+    defaultLanguage: string;
+    isEditing: boolean;
+    isPublicLink: boolean;
+    isAuthenticated: boolean;
+    isSearchDialogOpen: boolean;
+    languages: Language[];
+    template: TemplateContext | null;
+};
```
Comment (mike85) [inline packages/app-bridge-theme/src/types.ts:38]: Can we extract these types into a separate file?

File: packages/app-bridge-theme/src/types/Document.ts
Comment (mike85) [inline packages/app-bridge-theme/src/types/Document.ts:1]: We can move this into the `Guidelines.ts` file.

**Review by mike85 — CHANGES_REQUESTED** [review]

**Comment by Kenny806** [comment]
Closing in favour of #958

**Pull request closed without merging**